lib/middleman-images/manipulator.rb

```diff
@@ -6,30 +6,53 @@ def initialize(app, ignore_original)
         @images = []
         @required_originals = []
         @ignore_original = ignore_original
+        @inspected_at = {}
+        @destination_paths = []
       end
 
       def add(image)
-        unless images.collect(&:destination).include? image.destination
-          images << image
-        end
+        images << image unless @destination_paths.include?(image.destination_path)
+        @destination_paths << image.destination_path
       end
 
       def preserve_original(resource)
         required_originals << resource.source_file
       end
 
       def manipulate_resource_list(resources)
+        app.logger.info "== Images: Registering images to process. This may take a while…"
+
+        resources.each do |resource|
+          next unless inspect?(resource)
+
+          app.logger.debug "== Images: Inspecting #{resource.destination_path} for images to process."
+
+          @inspected_at[resource.source_file] = File.mtime(resource.source_file)
+          begin
+            resource.render({}, {})
```

Comment:
A comment here could help understand, why render is called (i.e. to ensure all image_tag helpers are called and resource objects are created - but not yet processed!).